noisy-simulator/src/instrument.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! This module contains the `Instrument` struct, used to make measurments
+//! in a quantum system.
+
+#[cfg(test)]
+mod tests;
+
+use crate::{operation::Operation, SquareMatrix, TOLERANCE};
+
+/// An instrument is the means by which we make measurements on a quantum system.
+pub struct Instrument {
+    operations: Vec<Operation>,
+    summed_operation: SquareMatrix,
+    summed_effect: SquareMatrix,
+    summed_effect_transpose: SquareMatrix,
+    summed_kraus_operators: Vec<SquareMatrix>,
+}
+
+impl Instrument {
+    /// Creates a new instrument.
+    pub fn new(operations: Vec<Operation>) -> Self {
+        let summed_operation: SquareMatrix = operations.iter().map(|op| op.matrix()).sum();
+        let summed_effect: SquareMatrix = operations.iter().map(|op| op.effect_matrix()).sum();
```

Comment:
Might be premature optimization, but 
```
let (summed_operations, summed_effect) = operations.iter().fold((<zero_mat>,<zero_mat>), |(sum_op, sum_eff), op| {
        (sum_op + op.matrix(), sum_eff + op.effect_matrix())
    });
```

Reply:
Why is this faster? Is it because it iterates once? Since these iterators usually have less than 10 elements, I don't think it will impact the performance.

Also, I don't know if the improvement would be noticeable compared to the matrix sums that are much slower.